nginx/datadog_checks/nginx/nginx.py

```diff
@@ -245,8 +238,14 @@ def _flatten_json(cls, metric_base, val, tags):
                 else:
                     tags = tags + [server]
             for key, val2 in val.iteritems():
-                metric_name = '%s.%s' % (metric_base, key)
-                output.extend(cls._flatten_json(metric_name, val2, tags))
+                if key in TAGGED_KEYS:
+                    metric_name = '%s.%s' % (metric_base, TAGGED_KEYS[key])
+                    for tag_val, data in val2.iteritems():
+                        tag = '%s:%s' % (TAGGED_KEYS[key], tag_val)
```

Comment:
I'd say instead `tags.append('%s:%s' % (TAGGED_KEYS[key], tag_val))`.

Reply:
After seeing the tests fail miserably, I remember why it was like this. I can't append to the `tags` list because since I am in a for loop, and `tags` is a reference to the list. So I have to do like this to avoid having the tags of a metric being sent with the metrics of all the subsequent loops.